### PR TITLE
Create test for H5Pget_dxpl_mpio

### PR DESCRIPTION
**Pull request by glennsong09**
Created a test for H5Pget_dxpl_mpio in testpar to ensure method works correctly.

### CODE REVIEW

File: testpar/t_ph5basic.c [hunk 1]
```diff
@@ -177,3 +177,119 @@ test_fapl_mpio_dup(void)
         VRFY((mrc == MPI_SUCCESS), "MPI_Info_free");
     }
 } /* end test_fapl_mpio_dup() */
+
+/*-------------------------------------------------------------------------
+ * Function:    test_get_dxpl_mpio
+ *
+ * Purpose:     Test that H5Pget_fxpl_mpio will properly return the data
```
Comment (jhendersonHDF) [inline testpar/t_ph5basic.c]: H5Pget_fxpl_mpio -> H5Pget_dxpl_mpio

File: testpar/t_ph5basic.c [hunk 2]
```diff
@@ -177,3 +177,119 @@ test_fapl_mpio_dup(void)
...
+    H5FD_mpio_xfer_t xfer_mode;
+    hsize_t          dims[2] = {100, 100};
+    hsize_t          i, j;
+    int             *data = NULL;
+    int              mpi_rank, mpi_size;
+    const char      *filename;
+    herr_t           ret;
+
+    if (VERBOSE_MED)
+        printf("Verify get_fxpl_mpio correctly gets the data transfer mode"
```
Comment (jhendersonHDF) [inline testpar/t_ph5basic.c]: H5Pget_fxpl_mpio -> H5Pget_dxpl_mpio

File: testpar/t_ph5basic.c [hunk 3]
```diff
@@ -177,3 +177,119 @@ test_fapl_mpio_dup(void)
...
+    ret = H5Pclose(fapl);
+    VRFY((ret >= 0), "H5Pclose succeeded");
+
+    ret = H5Pclose(dxpl);
+    VRFY((ret >= 0), "H5Pclose succeeded");
+
+    ret = H5Dclose(did);
+    VRFY((ret >= 0), "H5Dclose succeeded");
+
+    ret = H5Sclose(sid) < 0;
```
Comment (jhendersonHDF) [inline testpar/t_ph5basic.c]: Looks like the comparison to 0 here and below might be accidental leftovers

**Comment by jhendersonHDF** [comment]
It would be nice to also check the value after a call to H5Dread in both cases (collective vs. independent), but otherwise this looks reasonable

File: testpar/t_ph5basic.c (hunk 3 above)
Comment (glennsong09) [reply testpar/t_ph5basic.c]: Oh, let me fix that.

File: testpar/t_ph5basic.c (hunk 2 above)
Comment (glennsong09) [reply testpar/t_ph5basic.c]: Let me change these.

**Comment by glennsong09** [comment]
> It would be nice to also check the value after a call to H5Dread in both cases (collective vs. independent), but otherwise this looks reasonable

I can add that.

**Review by jhendersonHDF — APPROVED** [review]

**Review by brtnfld — APPROVED** [review]
lgtm

**Pull request merged**